### PR TITLE
[RHCLOUD-28007] Move Travis Check into Jenkins PR Check

### PR DESCRIPTION
**Pull request by Ellen-Yi-Dong**
## Link(s) to Jira
-https://issues.redhat.com/browse/RHCLOUD-28007

## Description of Intent of Change(s)
The what, why and how.

Travis is being deprecated, so we are moving the unit tests checks into the jenkins pipeline

## Local Testing
How can the feature be exercised?
How can the bug be exploited and fix confirmed?
Is any special local setup required?

## Checklist
- [ ] if API spec changes are required, is the spec updated?
- [ ] are there any pre/post merge actions required? if so, document here.
- [ ] are theses changes covered by unit tests?
- [ ] if warranted, are documentation changes accounted for?
- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?
- [ ] is there known, direct impact to dependent teams/components?
  - [ ] if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by petracihalova — APPROVED** [review]

**Comment by lpichler** [comment]
@Ellen-Yi-Dong

**Comment by lpichler** [comment]
@Ellen-Yi-Dong 
I appreciate your work on this pull request. I've noticed a high number of commits, some of which appear experimental and I totally understand why those commit are here(you needed to try it and test it) Consolidating these into fewer, more cohesive commits could enhance clarity and maintainability. I am suggesting to put those changes to few commits (I think it will be simpler to create new PR).

Let me know what do you think!

File: unit_test.sh [hunk 1]
```diff
@@ -1,27 +1,62 @@
...
+
+# Build PR_Check Container
+docker run -i --rm --name $CONTAINER_NAME \
+    -e DATABASE_NAME=$DATABASE_NAME \
+    -e DATABASE_HOST=$DATABASE_HOST \
+    -e DATABASE_PORT=$DATABASE_PORT \
+    -e DATABASE_USER=$DATABASE_USER \
+    -e DATABASE_PASSWORD=$DATABASE_PASSWORD \
+    --net=host \
+    $IMAGE_TAG tox
```
Comment (dagbay-rh) [inline unit_test.sh:41]: overall this looks great! makes sense its easier to get the db up and running in another local container vs having bonfire do it for us automagically

my only nit pick would be to add the `tox` command to the pr check dockerfile as a `CMD`. it doesnt make any real difference afaik vs running it like this, just reads a little nicer imo to have the intended command of the image in the dockefile itself

**Comment by dagbay-rh** [comment]
> @Ellen-Yi-Dong I appreciate your work on this pull request. I've noticed a high number of commits, some of which appear experimental and I totally understand why those commit are here(you needed to try it and test it) Consolidating these into fewer, more cohesive commits could enhance clarity and maintainability. I am suggesting to put those changes to few commits (I think it will be simpler to create new PR).
> 
> Let me know what do you think!

I agree here, condensing the commits before we merge would definitely be ideal. To add another option, I think we could simply squash when we merge, so we end up with just one nice commit in main. So long as we clean up the commit message I think that would work fine :cowboy_hat_face:

**Review by dagbay-rh — APPROVED** [review]

**Pull request merged**